mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/search/ElasticSearchSearcher.java

```diff
@@ -104,17 +104,29 @@ private SearchRequestBuilder getSearchRequestBuilder(Client client, List<User> u
     private Optional<SimpleMessageSearchIndex.SearchResult> extractContentFromHit(SearchHit hit) {
         SearchHitField mailboxId = hit.field(JsonMessageConstants.MAILBOX_ID);
         SearchHitField uid = hit.field(JsonMessageConstants.UID);
-        SearchHitField id = hit.field(JsonMessageConstants.ID);
-        if (mailboxId != null && uid != null && id != null) {
+        Optional<SearchHitField> id = retrieveIdField(hit);
+        if (mailboxId != null && uid != null) {
             Number uidAsNumber = uid.getValue();
             return Optional.of(
-                new MessageSearchIndex.SearchResult(com.google.common.base.Optional.of(messageIdFactory.fromString(id.getValue())),
+                new MessageSearchIndex.SearchResult(toGuava(id.map(field -> messageIdFactory.fromString(field.getValue()))),
                     mailboxIdFactory.fromString(mailboxId.getValue()),
                     MessageUid.of(uidAsNumber.longValue())));
         } else {
-            LOGGER.warn("Can not extract UID, MessageID and/or MailboxId for search result " + hit.getId());
+            LOGGER.warn("Can not extract UID and/or MailboxId for search result " + hit.getId());
             return Optional.empty();
         }
     }
 
+    private Optional<SearchHitField> retrieveIdField(SearchHit hit) {
+        if (hit.fields().keySet().contains(JsonMessageConstants.ID)) {
+            return Optional.ofNullable(hit.field(JsonMessageConstants.ID));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private <T> com.google.common.base.Optional<T> toGuava(Optional<T> optional) {
+        return com.google.common.base.Optional.fromNullable(optional.orElse(null));
```

Comment:
nope, ok (we could properly extract apache-james-backends-cassandra which depends on server-util8) but that's out of the scope of this pr